mailbox/cassandra/src/main/java/org/apache/james/mailbox/cassandra/mail/CassandraMailboxPathDAO.java

```diff
@@ -144,16 +147,38 @@ private PreparedStatement prepareSelectAll(Session session) {
                 .setUDTValue(NAMESPACE_AND_USER, mailboxBaseTupleUtil.createMailboxBaseUDT(mailboxPath.getNamespace(), mailboxPath.getUser()))
                 .setString(MAILBOX_NAME, mailboxPath.getName()))
             .thenApply(rowOptional ->
-                rowOptional.map(row -> new CassandraIdAndPath(
-                    CassandraId.of(row.getUUID(MAILBOX_ID)),
-                    mailboxPath)));
+                rowOptional.map(this::fromRowToCassandraIdAndPath))
+            .thenApply(value -> logGhostMailbox(mailboxPath, value));
     }
 
     public CompletableFuture<Stream<CassandraIdAndPath>> listUserMailboxes(String namespace, String user) {
         return cassandraAsyncExecutor.execute(
             selectAll.bind()
                 .setUDTValue(NAMESPACE_AND_USER, mailboxBaseTupleUtil.createMailboxBaseUDT(namespace, user)))
-            .thenApply(resultSet -> cassandraUtils.convertToStream(resultSet).map(this::fromRowToCassandraIdAndPath));
+            .thenApply(resultSet -> cassandraUtils.convertToStream(resultSet)
+                .map(this::fromRowToCassandraIdAndPath)
+                .peek(this::logReadSuccess));
+    }
+
+    public Optional<CassandraIdAndPath> logGhostMailbox(MailboxPath mailboxPath, Optional<CassandraIdAndPath> value) {
```

Comment:
idem the meaning is ok for us, but nobody reading this in a few month can understand it

Reply:
I'll add JavaDoc